fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingMultiTargetByRecords.java

```diff
@@ -202,8 +238,7 @@ CompletableFuture<Void> rebuildIndexInternalAsync(FDBRecordStore store) {
 
     @Nonnull
     @SuppressWarnings("PMD.CloseResource")
-    private CompletableFuture<Tuple> rebuildRangeOnly(@Nonnull FDBRecordStore store, Tuple cont, @Nonnull AtomicLong recordsScanned) {
-
+    private CompletableFuture<Tuple> rebuildRangeOnly(@Nonnull FDBRecordStore store, Tuple cont, @Nonnull AtomicLong recordsScanned, @Nullable Tuple rangeEnd) {
```

Comment:
This API doesn't seem right to me. I _think_ I understand what's going on, which is that it is assuming that index builds either require:

1. Scanning all records, in which case `rangeEnd` is `null` or
2. Scanning a set of range of records to cover all indexed types, in which case `rangeEnd` is the type key of the lexicographically last type

The problem I have with it is that the `rangeEnd` tuple here is a little too divorced from its origin for that to be clear to the user, especially because in most of the rest of the API, a bare `Tuple` is not assumed to be an inclusive right endpoint.

Reply:
Would a comment and renaming help?